couchable/core.py

```diff
@@ -1062,26 +1092,26 @@ def _unpack(self, parent_doc, doc, loaded_dict, inst=None):
                     elif method_str == 'pickle':
                         if 'pickles' not in parent_doc[FIELD_NAME]:
                             attachment_response = self.db.get_attachment(parent_doc, 'pickles')
-                            parent_doc[FIELD_NAME]['pickles'] = pickle.loads(doGunzip(attachment_response.read()))
+                            parent_doc[FIELD_NAME]['pickles'] = doUnpickle(
+                                doGunzip(attachment_response.read())
+                            )
                             #parent_doc[FIELD_NAME]['pickles'] = collections.defaultdict(int)
 
                         return parent_doc[FIELD_NAME]['pickles'][data]
 
                     type_str, data = data.split(':', 1)
                     if method_str == 'append':
-                        if type_str == 'unicode':
-                            if isinstance(data, unicode):
-                                return data
+                        if type_str in ('unicode', 'str'):
+                            if isinstance(data, bytes):
+                                return data.decode('UTF-8')
                             else:
-                                return unicode(data, 'utf8')
-                        if type_str == 'str':
-                            return str(data)
+                                return data
 
                     elif method_str == 'repr':
-                        if type_str in __builtins__:
-                            return __builtins__.get(type_str)(data)
-                        elif type_str == '__builtin__.NoneType':
+                        if type_str == 'NoneType':
```

Comment:
Shadowed by above comment: `__builtins__` is an implementation detail of CPython and should have never been used on the first place. Using `hasattr` and `getattr` on `builtins` is the correct approach and portable between implementations and Python versions. For `None` we need to check for the type string (`'NoneType'`) and directly return `None`, since `None` is a global singleton.